### DISCUSSION TITLE
showing relationships in list operation?

### DISCUSSION THREAD

**Discussion by prescriptionlifeline** [Laravel-Backpack/community-forum#1038] [Q&A (Help)] [answered]
In `setupCreateOperation`, in my User CRUD, I have this:

```
        CRUD::field([
            'name' => 'currentDues',
            'label' => date('Y') . ' Dues',
            'type' => 'relationship',
            'subfields' => [
                [
                    'label' => 'Date Paid',
                    'name' => 'datePaid',
                    'type' => 'date',
                ],
                [
                    'label' => 'Amount Paid',
                    'name' => 'amountPaid',
                    'type' => 'number',
                ],
                [
                    'name' => 'year',
                    'type' => 'hidden',
                    'value' => date('Y'),
                ],
            ],
        ]);
```
Here is the `currentDues` method in the User model:

```
    public function currentDues()
    {
        return $this->hasOne(Dues::class, 'userId')->where('year', date('Y'));
    }
```

I'd like to show that as two distinct columns in `setupListOperation()`.

One thought I have is to create a new `amountDueThisYear` column in my users table in the DB and add this method to my `Dues` model (untested):

```
    protected static function booted()
    {
        $updateDues = function ($due) {
            $user = User::findOrFail($due->userId);
            $user->amountDueThisYear = $due->amountDue;
            $user->save();
        };
        static::creating($updateDues);
        static::updating($updateDues);
    }
```
I guess I could create an SQL TRIGGER to auto-update these newly created columns in the users table as well but the `booted()` method seems preferable to that.

Are these my best options or is there a more straight forward way to do this that I'm not seeing?

**Comment by pxpm** [comment, marked as answer]
Hello @prescriptionlifeline 

if you have your data loaded in the model (as part of a database column, or you manually add it with `$appends` https://laravel.com/docs/11.x/eloquent-serialization#appending-values-to-json ), you can just do: 
```php
CRUD::column('currentDues.amountPaid');
CRUD::column('currentDues.yourComputedColumnName');
```

Another thing that I highly advise you to avoid having unexpected issues, or having to do additional configurations to the framework, it to use the conventions. 
Laravel conventions is to name `relation_id`, so `user_id` instead of `userId`. 

Cheers
